tests/base.py

```diff
@@ -0,0 +1,388 @@
+import unittest
+import os
+from datetime import timedelta
+from datetime import datetime as dt
+import dateutil.parser
+import pytz
+from typing import Dict, Set
+from tap_tester import connections, menagerie, runner
+from tap_tester.logger import LOGGER
+
+class YotpoBaseTest(unittest.TestCase):
+
+    PRIMARY_KEYS = "table-key-properties"
+    START_DATE_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+    BOOKMARK_COMPARISON_FORMAT = "%Y-%m-%dT00:00:00+00:00"
+    REPLICATION_KEYS = "valid-replication-keys"
+    REPLICATION_METHOD = "forced-replication-method"
+    INCREMENTAL = "INCREMENTAL"
+    FULL_TABLE = "FULL_TABLE"
+    STARTDATE_KEYS = "start_date_key"
+    DATETIME_FMT = {
+        "%Y-%m-%dT%H:%M:%SZ",
+        "%Y-%m-%d %H:%M:%S",
+        "%Y-%m-%dT%H:%M:%S.000000Z"
+    }
+
+    def expected_replication_method(self):
+        """return a dictionary with key of table name nd value of replication method"""
+        return {table: properties.get(self.REPLICATION_METHOD, None)
+                for table, properties
+                in self.expected_metadata().items()}
+
+    def setUp(self):
+        missing_envs = [x for x in [os.getenv('TAP_YOTPO_API_KEY'),
+                                    os.getenv('TAP_YOTPO_API_SECRET')] if x == None]
+        if len(missing_envs) != 0:
+            raise Exception("Missing test-required environment variables")
+
+    def get_type(self):
+        return "platform.yotpo"
+
+    @staticmethod
+    def tap_name():
+        return "yotpo"
+
+    def expected_metadata(self):
+        """The expected streams and metadata about the streams"""
+        return {
+            'emails': {
+                self.PRIMARY_KEYS: {'email_address', 'email_sent_timestamp'},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {'email_sent_timestamp'}
```

Comment:
Use the OBEY_START_DATE key here to define if the stream is following start_date or not, and use this in the start_date test. We got this comment in the past, so following it for all the taps now.
Reference: https://github.com/singer-io/tap-github/pull/167/files#diff-bdbf5290b28c74db41a6af29864daeabbed3fcc92051c15111aebaf5ba5ba703R155